flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManager.java

```diff
@@ -585,7 +585,8 @@ private TaskManagerSlot createAndRegisterTaskManagerSlot(SlotID slotId, Resource
 	@Nullable
 	private PendingTaskManagerSlot findExactlyMatchingPendingTaskManagerSlot(ResourceProfile resourceProfile) {
 		for (PendingTaskManagerSlot pendingTaskManagerSlot : pendingSlots.values()) {
-			if (pendingTaskManagerSlot.getResourceProfile().equals(resourceProfile)) {
+			// there might be slight error between slot resource profiles calculated by RM / TM
+			if (pendingTaskManagerSlot.getResourceProfile().approximate(resourceProfile)) {
```

Comment:
I'm a little worry that the approximation may not be tolerant in TM side. For example, if the network memory of the available slot is just 1MB less than the request slot, it is very likely to cause the InsufficientBufferException in the TM side. 

I'm not sure why we can't make the request exactly the same with the available slot ? Or at least we must ensure pendingTaskManagerSlot.getResourceProfile().isMatching(resourceProfile) is also true.

Reply:
The reason we need approximation is due to precision issue. Currently slot profile are calculated on both RM and TM, with the same rules but different implementations. As a result, the calculation results may have slight difference. We are planing to unify the memory calculation so that RM and TM use the same implementation in the next version. Once we have that, the approximation could be removed.

Reply:
Why does the RM re-calculate the resource profile? This assumes that the RM and TM not only follow the same rules, but have the same config. That is actually not always the case, I have seen standalone setups where different TMs had different configs.

Reply:
As we discussed offline, I reverted changes here.